website/client/js/controllers/settingsCtrl.js

```diff
@@ -287,6 +293,33 @@ habitrpg.controller('SettingsCtrl',
       return Math.floor(numberOfHourglasses);
     };
 
+    $scope.hasSocialAuth = function(user) {
+      SOCIAL_AUTH_NETWORKS.forEach(function (network) {
+        if (user.auth.hasOwnProperty(network.key)) {
+          return Boolean(user.auth[network.key].id)
```

Comment:
If this can ever return `false`, it would be possible to return `false` before a potential `true` is reached.